pallets/parachain-staking/src/lib.rs

```diff
@@ -438,9 +442,50 @@ pub mod pallet {
 				weight = weight.saturating_add(T::OnNewRound::on_new_round(round.current));
 				// pay all stakers for T::RewardPaymentDelay rounds ago
 				weight = weight.saturating_add(Self::prepare_staking_payouts(round.current));
+
+				// select total collators
+				let collators = <SelectedCandidates<T>>::get();
+
+				// if collators length is below or eq to 66% of max_collators,
+				// we don't mark any other collator as offline
+				let max_collators = <TotalSelected<T>>::get();
+
+				// track collators length
+				let mut len_counter = collators.len();
+
+				// iter collators to check which of them must be marked as offline
+				for collator in collators {
+					if let Some(info) = <CandidateLastActive<T>>::get(&collator) {
+						if round.current.saturating_sub(info.last_round)
+							> T::MaxOfflineRounds::get() && len_counter * 3
+							> (max_collators * 2) as usize
+						{
+							// if the collator has not produced any block within
+							// MaxOfflineRounds e.g(3 rounds for Moonriver)
+							// it is marked as offline
+							Self::do_go_offline(collator.clone()).unwrap_or_default();
```

Comment:
```suggestion
							let _ = Self::do_go_offline(collator.clone());
```

You don't need to transform the return value here